amp/parse.go

```diff
@@ -32,6 +41,62 @@ type Size struct {
 	Width          int64
 }
 
+// Policy consent types
+const (
+	ConsentDenied = iota
+	ConsentTCF1
+	ConsentTCF2
+	ConsentUSPrivacy
+)
+
+func ReadPolicy(ampParams Params, req *openrtb2.BidRequest, pbsConfigGDPREnabled bool) (privacy.PolicyWriter, error) {
+
+	if len(ampParams.Consent) == 0 {
+		return privacy.NilPolicyWriter{}, nil
+	}
+
+	var rv privacy.PolicyWriter = privacy.NilPolicyWriter{}
+	var warning error
+	var errMsg string
+
+	switch ampParams.ConsentType {
+	case ConsentTCF2:
+		if pbsConfigGDPREnabled {
+			// Even if consent is invalid, we write the consent string to req.user.ext.consent
+			rv = gdpr.ConsentWriter{ampParams.Consent}
+
+			if tcf2.IsConsentV2(ampParams.Consent) {
+				if _, err := tcf2.ParseString(ampParams.Consent); err != nil {
+					errMsg = err.Error()
+				}
+			} else {
+				errMsg = fmt.Sprintf("Consent string '%s' is not a valid TCF2 consent string.", ampParams.Consent)
+			}
+		}
+	case ConsentUSPrivacy:
+		if ccpa.ValidateConsent(ampParams.Consent) {
+			rv = ccpa.ConsentWriter{ampParams.Consent}
+		} else {
+			errMsg = fmt.Sprintf("Consent string '%s' is not a valid CCPA consent string.", ampParams.Consent)
+		}
+	case ConsentDenied:
+		errMsg = "Consent denied. Consent string ignored."
```

Comment:
I'm curious, what does consent denied mean?

Reply:
Renamed to `ConsentNone`